config/v1/types_cluster_version.go

```diff
@@ -406,17 +432,34 @@ type ComponentOverride struct {
 type URL string
 
 // Update represents an administrator update request.
+// +kubebuilder:validation:XValidation:rule="has(self.architecture) && has(self.image) ? (size(self.architecture) == 0 || size(self.image) == 0) : true",message="cannot set both Architecture and Image"
+// +kubebuilder:validation:XValidation:rule="has(self.architecture) && size(self.architecture) != 0 ? size(self.version) != 0 : true",message="Version must be set if Architecture is set"
```

Comment:
Can we simplify (and use cluster-user-facing downcased JSON/YAML property names) with:

```go
// +kubebuilder:validation:XValidation:rule="size(self.architecture) > 0 ? size(self.image) == 0,message="cannot set both architecture and image"
// +kubebuilder:validation:XValidation:rule="size(self.architecture) > 0 ? size(self.version) > 0 : true",message="version must be set if architecture is set"
```

Reply:
If you don't check that the property exists with `has` first and it's not there `size` gets `no such key` errors.

Reply:
These are not pointer properties.  What does "it's not there" mean?  Is that "`self` is `nil`" when the whole `desiredUpdate` is unset?

Reply:
When CEL validation occurs, it uses an unstructured object during the validation so it only knows about fields that are represented in the `yaml` that is sent to the API. That means that there is a possibility that fields are not present, which then causes the validation to fail.

Having the `has` check is the correct way to prevent errors for fields that are optional in our APIs.

Reply:
I know it fails the test cases when not there. My thought is that it's because they're optional, and therefore not there, checking them with anything like `size` causes a failure.